org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Finally.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+import static org.jacoco.core.test.validation.targets.Stubs.nop;
+
+/**
+ * This test target is a finally block.
+ */
+public class Finally {
```

Comment:
Just nested: A try/finally block within a finally block. [This](https://github.com/marchof/will-this-compile/blob/master/will-this-compile/src/NestedFinally.java) is probably too much 😉 .

Also a test for finally blocks with some control structures (e.g. `if`) would be a nice test for the `isSame()` method.